Source/JavaScriptCore/heap/BlockDirectory.cpp

```diff
@@ -388,6 +388,16 @@ void BlockDirectory::shrink()
     }
 }
 
+MarkedBlock::Handle* BlockDirectory::findMarkedBlockHandle(MarkedBlock* block)
+{
+    for (size_t index = 0; index < m_blocks.size(); ++index) {
+        MarkedBlock::Handle* handle = m_blocks[index];
+        if (handle->hasBlock(block))
```

Comment:
rather than introduce hasBlock() (which should be removed later with all the other debugging code as it doesn't really make sense as a standalone interface), why not just do:

if (block == &handle->block())